### PR TITLE
EZP-29106: Impl. "Accept" header overriding

### PR DESCRIPTION
**Pull request by adamwojs**
> JIRA: https://jira.ez.no/browse/EZP-29106

## Description 

This PR introduces possibility to override `Accept` header for all `GET` requests.

## TODO: 

- [x] Implement tests.

### CODE REVIEW

**Comment by sunpietro** [comment]
I'm missing use cases for this change

**Comment by adamwojs** [comment]
@sunpietro See related PR: https://github.com/ezsystems/PlatformUIBundle/pull/964

**Comment by adamwojs** [comment]
I've added `acceptHeader` as the last one parameter in each method for two reasons:
* Is optional parameter 
* It's simplify things a lot: I don't have to check if method was called with or without `acceptHeader` as in follows example:

https://github.com/ezsystems/ez-js-rest-client/blob/5f889a6e71b84fcfaac61bcc66d743b24dd6c987/src/services/ContentService.js#L573-L576

**Comment by sunpietro** [comment]
@adamwojs regarding this repository, we have to follow the same convention as everywhere, so the callback should come last. I don't like it, but it has to be consistent.

**Comment by adamwojs** [comment]
PR updated according to comments.

**Comment by andrerom** [comment]
>  I don't like it, but it has to be consistent.

@sunpietro can you expand on "has to"? I never understood why it has to be the last argument actually :)

**Comment by adamwojs** [comment]
> could you add on ContentService.prototype.createView as well?

I think it's possible. Maybe we should be and add this possibility to other requests as well (for consistency) ?

> should there be changes ti tests for this?

Yes, off course.

**Comment by sunpietro** [comment]
@andrerom Neither I, but it was the rule that came from the previous repository maintainer. Personally, I prefer adding new params at the end of the list of params. It's easier for development and maintaining.
We can break this rule and start using a new approach from now on.
Also, tests are required @adamwojs

File: src/services/ContentTypeService.js [hunk 1]
```diff
@@ -179,15 +185,21 @@ define(["structures/ContentTypeGroupInputStruct", "structures/ContentTypeCreateS
      *
      * @method loadContentTypeGroup
      * @param contentTypeGroupId {String} target content type group identifier (e.g. "/api/ezp/v2/content/types/100")
+     * @param [acceptHeader] {String} Optional accept header value
      * @param callback {Function} callback executed after performing the request (see
      *  {{#crossLink "ContentTypeService"}}Note on the callbacks usage{{/crossLink}} for more info)
      */
-    ContentTypeService.prototype.loadContentTypeGroup = function (contentTypeGroupId, callback) {
+    ContentTypeService.prototype.loadContentTypeGroup = function (contentTypeGroupId, acceptHeader, callback) {
+        if ( !callback ) {
```
Comment (sunpietro) [inline src/services/ContentTypeService.js]: `if (!callback) {`

Comment (sunpietro) [reply src/services/ContentTypeService.js]: Here and in other places like this.

**Comment by andrerom** [comment]
> We can break this rule and start using a new approach from now on.

I'm open to either, was hoping someone might know why callback is last, like if that is needed for some callback related logic 


> I think it's possible. Maybe we should be and add this possibility to other requests as well (for consistency) ?

@adamwojs isn't it mostly the only one missing by now? :)

**Comment by adamwojs** [comment]
> @adamwojs isn't it mostly the only one missing by now? :)

At the moment all non `GET` requests doesn't allow to override Accept header. `create View` using `POST`.

https://github.com/ezsystems/ez-js-rest-client/blob/6f51f291e91229c236d48b8e3ec9a68466323bbf/src/services/ContentService.js#L1248-L1254

But I think in this case is already possible to set `Accept` header to custom value: 

```js
query = contentService.newViewCreateStruct(/* ... */);
query.headers['Accept'] = 'application/vnd.acme.View+json';
// ... 
contentService.createView(query, callback);
```

As I right @sunpietro ?

**Comment by adamwojs** [comment]
PR updated according to suggestions.

**Comment by andrerom** [comment]
> But I think in this case is already possible to set Accept header to custom value:

On second look I think so as well

**Pull request merged**